lib/service/storage/elasticsearch.js

```diff
@@ -2362,10 +2362,49 @@ class ElasticSearch extends Service {
 
     return parsedValue;
   }
+
+  /**
+   * Returns true if one of the mappings dynamic property changes value from
+   * false to true
+   */
+  _dynamicChanges (previousMappings, newMappings) {
+    const previousValues = findDynamic(previousMappings);
+
+    for (const [path, previousValue] of Object.entries(previousValues)) {
+      if (previousValue.toString() !== 'false') {
+        continue;
+      }
+
+      const newValue = _.get(newMappings, path);
+
+      if (newValue && newValue.toString() === 'true') {
```

Comment:
`dynamic` can be one of the following values: `false, true, strict`

I think that we should also trigger a reindex if the dynamic value goes from false to strict.